### PR TITLE
[resolves #416] Support saving RealtimeChart as GIF file

### LINKED ISSUE #416: How saving real time charts?
How can I save real-time charts as GIF?
I added this at the end of my code but it only saves the chart with the last update of the XYSeries:
`BitmapEncoder.saveBitmap(this.chart, "./Sample_Chart", BitmapFormat.GIF);`

### PR DESCRIPTION
**Pull request by Mr14huashao**
Add dependent animated-gif-lib, create git file
```xml
<dependency>
  <groupId>com.madgag</groupId>
  <artifactId>animated-gif-lib</artifactId>
</dependency>
```
license is Apache-2.0 license

RealtimeChart save as gif file:
RealtimeChart_GIF

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -79,6 +79,11 @@
 				<version>0.25</version>
 				<optional>true</optional>
 			</dependency>
+			<dependency>
```
Comment (timmolter) [inline pom.xml]: This should be optional.

Comment (Mr14huashao) [reply pom.xml]: already edited

File: xchart/src/main/java/org/knowm/xchart/GifOperator.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import org.knowm.xchart.internal.Utils;
+
+import com.madgag.gif.fmsware.AnimatedGifEncoder;
+
+/**
+ * A helper class with static methods for saving Charts as a GIF file
+ *
+ * @author Mr14huashao
+ */
+public class GifOperator {
```
Comment (timmolter) [inline xchart/src/main/java/org/knowm/xchart/GifOperator.java]: Please rename this `GifEncoder`.

Comment (Mr14huashao) [reply xchart/src/main/java/org/knowm/xchart/GifOperator.java]: already edited

**Pull request merged**